### PR TITLE
Added flake8 test and updated affected code

### PR DESCRIPTION
**Pull request by joshthoward**
Added flake8 to the GitHub CI tests and then updated the code to pass flake8 test. Some unused imports were removed.

The max line length is set to 120 and the W503 rule is ignored which is PEP 8 compliant.

### CODE REVIEW

**Comment by joshthoward** [comment]
CLA has been sent via email.

File: integration_tests/test_dbapi.py [hunk 1]
```diff
@@ -171,12 +162,12 @@ def test_boolean_query_param(trino_connection):
     cur.execute("SELECT ?", params=(True,))
     rows = cur.fetchall()
 
-    assert rows[0][0] == True
+    assert rows[0][0]
```
Comment (hashhar) [inline integration_tests/test_dbapi.py:165]: I wouldn't consider this equivalent. Can lead to silent bugs when copying tests.

This change is only safe because you know that the type is a boolean. Truthiness is different for other types.

Same at L170.

File: integration_tests/test_dbapi.py [hunk 2]
```diff
@@ -201,6 +192,7 @@ def test_float_nan_query_param(trino_connection):
 
 @pytest.mark.skip(reason="Nan currently not returning the correct python type fon inf")
 def test_float_inf_query_param(trino_connection):
+    cur = trino_connection.cursor()
```
Comment (hashhar) [inline integration_tests/test_dbapi.py:195]: Thanks. ❤️

File: integration_tests/test_dbapi.py [hunk 3]
```diff
@@ -368,15 +360,15 @@ def test_transaction_multiple(trino_connection_with_transaction):
...
-    """
-    tests that an invalid query raises the correct exception
+    """Tests that an invalid query raises the correct exception
     """
     cur = trino_connection.cursor()
     with pytest.raises(TrinoQueryError):
         cur.execute(
             """
-            select * FRMO foo WHERE x = ?; 
+            select * FRMO foo WHERE x = ?;
```
Comment (hashhar) [inline integration_tests/test_dbapi.py:371]: While we are at it, can you make SQL queries use uppercase keywords (or at-least be consistent within a single query) if it's not too much of an effort?

It should be a separate commit though.

File: setup.cfg [hunk 4]
```diff
@@ -1,2 +1,6 @@
 [aliases]
 test=pytest
+
+[flake8]
+max-line-length = 120
+ignore = W503
```
Comment (hashhar) [inline setup.cfg:6]: Please add a comment about why this rule is ignored (and what the rule says).

File: tests/test_client.py [hunk 5]
```diff
@@ -214,8 +213,8 @@
                 "com.facebook.presto.sql.tree.Table.accept(Table.java:50)",
                 "com.facebook.presto.sql.tree.AstVisitor.process(AstVisitor.java:22)",
                 "com.facebook.presto.sql.analyzer.StatementAnalyzer.analyzeFrom(StatementAnalyzer.java:1413)",
-                "com.facebook.presto.sql.analyzer.StatementAnalyzer.visitQuerySpecification(StatementAnalyzer.java:670)",
-                "com.facebook.presto.sql.analyzer.StatementAnalyzer.visitQuerySpecification(StatementAnalyzer.java:166)",
+                "com.facebook.presto.sql.analyzer.StatementAnalyzer.visitQuerySpecification(StatementAnalyzer.java:670)",  # NOQA
```
Comment (hashhar) [inline tests/test_client.py:216]: Is it possible to do this more cleanly? For the entire block at once?

File: trino/__init__.py [hunk 6]
```diff
@@ -20,4 +20,6 @@
 from . import exceptions
 from . import logging
 
+__all__ = ['auth', 'dbapi', 'client', 'constants', 'exceptions', 'logging']
```
Comment (hashhar) [inline trino/__init__.py:23]: I'm not sure if we consider all of these to be public API.

Most notably `client` isn't supposed to be public since it's an implementation detail of the driver. People can use it but it's not expected to be stable (which is what `__all__` indicates by convention.

`auth`, `dbapi`, `constants` and `exceptions` might be considered public.

File: trino/dbapi.py [hunk 7]
```diff
@@ -23,16 +23,9 @@
 
 from typing import Any, List, Optional  # NOQA for mypy types
 
-try:
-    from urllib.parse import urlencode
-except ImportError:
-    # Python 2
-    from urllib import urlencode
```
Comment (hashhar) [inline trino/dbapi.py:30]: Was this unused?

This drops Python 2 support. The sad truth is that many people are still on 2.7.

File: integration_tests/conftest.py [hunk 8]
```diff
@@ -34,14 +34,6 @@
 TRINO_PORT = 8080
 
 
-def is_process_alive(pid):
```
Comment (hashhar) [inline integration_tests/conftest.py:37]: It's surprising that this is unused. I'll check what changed to make this unused.

**Review by hashhar — COMMENTED** [review]
A few comments.

Thanks for your contribution. ❤️

File: trino/dbapi.py (hunk 7 above)
Comment (findepi) [reply trino/dbapi.py:30]: @hashhar per the diff, looks like unused import?

File: tests/test_client.py (hunk 5 above)
Comment (joshthoward) [reply tests/test_client.py]: Not to my knowledge. You can make this cleaner on line continuations, but not for arrays. Another option would be for the whole file, but I don't think that is better at this point. 

I am going to update this to only remove checks for line length though.

File: trino/__init__.py (hunk 6 above)
Comment (joshthoward) [reply trino/__init__.py:23]: `__all__` is updated here to indicate to flake8 that the unused imports are part of a public API. Otherwise, we might want to remove `client` and `logging` from `__init__.py`. What do you think?

File: trino/dbapi.py (hunk 7 above)
Comment (joshthoward) [reply trino/dbapi.py:30]: It was unused.

File: integration_tests/test_dbapi.py [hunk 9]
```diff
@@ -49,11 +47,11 @@ def trino_connection_with_transaction(run_trino):
 
 def test_select_query(trino_connection):
     cur = trino_connection.cursor()
-    cur.execute("select * from system.runtime.nodes")
+    cur.execute("SELECT * FROM system.runtime.nodes")
```
Comment (findepi) [inline integration_tests/test_dbapi.py:50]: is this being checked too?

File: setup.cfg [hunk 10]
```diff
@@ -1,2 +1,11 @@
 [aliases]
 test=pytest
+
+[flake8]
+max-line-length = 120
+# W503 raises a warning when there is a line break before a binary operator.
+# This is best practice according to PEP 8 and the rule should be ignored. 
+#
+# https:#www.flake8rules.com/rules/W503.html
+# https:#www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator
```
Comment (findepi) [inline setup.cfg:10]: # -> // in url

File: trino/__init__.py (hunk 6 above)
Comment (hashhar) [reply trino/__init__.py:23]: Good point. Let's leave it out of this PR - I'll follow up on this in https://github.com/trinodb/trino-python-client/issues/43

File: integration_tests/conftest.py (hunk 8 above)
Comment (hashhar) [reply integration_tests/conftest.py:37]: This function was never called but has been here since the beginning. 🤷‍♂️ 

Let's remove it then. 👍

**Review by hashhar — APPROVED** [review]
LGTM 🚀

File: integration_tests/test_dbapi.py (hunk 9 above)
Comment (joshthoward) [reply integration_tests/test_dbapi.py:50]: It is not since it isn't "Python" code.

File: setup.cfg (hunk 10 above)
Comment (joshthoward) [reply setup.cfg]: Fixed.

**Comment by joshthoward** [comment]
@findepi, @hashhar can either of you guys merge my PR? I think we should be good to close here.

File: setup.cfg [hunk 11]
```diff
@@ -6,6 +6,6 @@ max-line-length = 120
 # W503 raises a warning when there is a line break before a binary operator.
 # This is best practice according to PEP 8 and the rule should be ignored. 
 #
-# https:#www.flake8rules.com/rules/W503.html
-# https:#www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator
+# https://www.flake8rules.com/rules/W503.html
+# https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator
```
Comment (findepi) [inline setup.cfg:10]: This should be squashed into first commit.

**Review by findepi — APPROVED** [review]
LGTM!

One minor comment about commit boundaries.

File: setup.cfg (hunk 11 above)
Comment (joshthoward) [reply setup.cfg:10]: @findepi I was being lazy 😔 This is fixed now.

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
Merged, thanks!
